src/operators.jl

```diff
@@ -328,119 +328,44 @@ end
 ###############################################################################
 # convenience/utility definitions
 
-const GenericAffOrQuadExpr = Union{GenericAffExpr, GenericQuadExpr}
-
-_densify_with_jump_eltype(x::AbstractMatrix) = convert(Matrix, x)
-
-function _densify_with_jump_eltype(x::SparseMatrixCSC{V}) where {V <: AbstractVariableRef}
-    return convert(Matrix{GenericAffExpr{Float64, V}}, x)
-end
-
-# See https://github.com/JuliaLang/julia/pull/18218.
-_A_mul_B_eltype(::Type{R}, ::Type{S}) where {R, S} = typeof(one(R) * one(S) + one(R) * one(S))
-
-_A_mul_B_ret_dims(A::AbstractMatrix, B::AbstractVector) = (size(A, 1),)
-_A_mul_B_ret_dims(A::AbstractMatrix, B::AbstractMatrix) = (size(A, 1), size(B, 2))
-
-# Don't do size checks here; defer that to `_A_mul_B(A, B)`.
-function _A_mul_B_ret(A, B, dims...)
-    T = _A_mul_B_eltype(eltype(A), eltype(B))
-    ret = Array{T}(undef, _A_mul_B_ret_dims(A, B))
-    return _fill_with_zeros!(ret, T)
-end
-
-function _fill_with_zeros!(A, ::Type{T}) where {T}
+function _fill_with_zeros!(A)
     for I in eachindex(A)
-        A[I] = zero(T)
+        A[I] = zero(eltype(A))
     end
     return A
 end
 
 ###############################################################################
-# `_A_mul_B!(ret, A, B)` adds the result of `A*B` into the buffer `ret`. We roll our own
+# `_mul!(ret, A, B)` adds the result of `A*B` into the buffer `ret`. We roll our own
```

Comment:
s/adds/sets/. The difference is `ret = A*B` vs `ret += A*B`.

Reply:
Good point